### DISCUSSION TITLE
Proxy loses connection and stops broadcasting MQTT messages when it's dark?

### DISCUSSION THREAD

**Discussion by vibejamin** [s-allius/tsun-gen3-proxy#208] [Q&A]
Hello,

first of all let me thank @s-allius for this great project! 

Although it was my first experience working with Docker I finally managed to get it up and running and it's really great for keeping an eye on all the different parameters the TSUN app doesn't let you see (i.e. voltages and currents per PV module etc.).

I'm running tsun-proxy - 0.11.0+dev and have noticed some of the "Connection reset by peer" messages @sellmann mentioned in his post on 12th June. I'm currently checking if it automatically reconnects after a while which it should as far as I've understood the replies.

What I've also noticed is that the proxy stops broadcasting MQTT messages as soon as the inverter stops generating power due to loss of sunlight. 

Is this intended behaviour? I noticed it when I played around with MQTT Explorer last evening and couldn't see anything from the proxy. But as soon as I checked in the morning it was all back there and even reconnected to the TSUN cloud.

If you need any more information I will be happy to provide.

**Comment by s-allius** [comment]
Hi crapforge,

yes this is the intended behaviour. The inverter disconnects the connection, wheeryn the it is dark. With my MS2000 inverter I get a few packages with the power of 0W, before the inverter shuts down and my MS800 inverter disconnects if the power is less than 2W.

Every time when the inverter has no power the logging module will close the TCP connection and you see the "Connection reset by peer". If the inverter doesn't have enough power for a graceful close, you will get a timeout error after a few minutes from the proxy. 

So everything seems to work fine in your setup.

Congratulation for you first Docker project.
